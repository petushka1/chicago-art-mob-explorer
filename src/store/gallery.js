import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';

const baseURL = 'https://api.artic.edu/api/v1/';
const artists = 'artists';
const artworks = 'artworks';
const exhibitions = 'exhibitions';
// search request by default cats
const search = 'https://api.artic.edu/api/v1/artworks/search?q=cats';
const LIMIT = 100;

export const fetchArtworks = createAsyncThunk(
  'gallery/fetchArtworks',
  (page = 1) => fetch(`${baseURL}${artworks}?page=${page}&limit=${LIMIT}`)
    .then((res) => res.json())
    .then((data) => data),
);

export const fetchExhibitions = createAsyncThunk(
  'gallery/fetchExhibitions',
  (page) => fetch(`${baseURL}${exhibitions}?page=${page}&limit=${LIMIT}`)
    .then((res) => res.json())
    .then((data) => data),
);

const gallerySlice = createSlice({
  name: 'gallery',
  initialState: [[], 0],
  reducers: {

  },

  extraReducers: {
    [fetchArtworks.fulfilled]: (state, { payload }) => {
      const { data } = payload;
      console.log(data);
      const newState = data.map((item) => (
        {
          id: item.id,
          image_id: item.image_id,
          artist_id: item.artist_id,
          category_ids: item.category_ids,
          artwork_type_id: item.artwork_type_id,
          category_titles: item.category_titles,
          title: item.title,
          artist_title: item.artist_title,
          date_start: item.date_start,
          date_end: item.date_end,
          place_of_origin: item.place_of_origin,
          artwork_type_title: item.artwork_type_title,
          technique_titles: item.technique_titles,
          material_titles: item.material_titles,
          medium_display: item.medium_display,
          thumbnail: item.thumbnail.lqip,
        }));
      return newState;
    },

    [fetchExhibitions.fulfilled]: (state, { payload }) => {
      const { data, pagination } = payload;
      const newState = data.filter((item) => item.artist_ids.length > 0).map((item) => ({
        id: item.id,
        title: item.title,
        image_url: item.image_url,
        status: item.status,
        artwork_ids: item.artwork_ids,
        department_display: item.department_display,
        artist_ids: item.artist_ids,
      }));
      console.log(newState);
      const { total } = pagination;
      return [newState, total];
    },

  },
});

export const galleryReduser = gallerySlice.actions;
export default gallerySlice.reducer;
