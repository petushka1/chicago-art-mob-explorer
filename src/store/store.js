import { configureStore } from '@reduxjs/toolkit';
import gallerySlice from './gallery';

const Store = configureStore({
  reducer: {
    gallery: gallerySlice,
  },
});

export default Store;
