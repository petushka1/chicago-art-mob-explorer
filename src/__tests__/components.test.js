/* eslint-disable */

import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Store from '../store/store';
import Header from '../components/header';
import { fetchExhibitions } from '../store/gallery';

  it('Renders Header Correctly', () => {
    const payload = {
      pagination: {
        total: 6165,
      },
      data: [
        {
          id: '1',
          title: 'Title 1',
          image_url: 'url',
          status: 'closed',
          artwork_ids: [12, 3435, 4, 898],
          department_display: 'some Dep',
          artist_ids: [1, 2],
        },
        {
          id: '2',
          title: 'Title 2',
          image_url: 'url',
          status: 'closed',
          artwork_ids: [12, 3435, 4, 898],
          department_display: 'some Dep',
          artist_ids: [1, 2],
        },
        {
          id: '3',
          title: 'Title 2',
          image_url: 'url',
          status: 'closed',
          artwork_ids: [12, 3435, 4, 898],
          department_display: 'some Dep',
          artist_ids: [1, 2],
        },
      ],
    };

    global.fetch = jest.fn(
      () => Promise.resolve({ json: () => Promise.resolve(payload) }),
    );
    Store.dispatch(fetchExhibitions('gallery/fetchExhibitions'));

    const header = render(
      <Provider store={Store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );

    expect(header).toMatchSnapshot();
  });
