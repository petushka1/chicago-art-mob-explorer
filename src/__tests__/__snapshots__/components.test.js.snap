// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Header Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="headerWrapper"
      >
        <div
          class="color"
        >
          <div
            class="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              <img
                alt="icon"
                class="icon"
                src="home.svg"
              />
            </a>
          </div>
          <div
            class="header"
          >
            <div
              class="logo"
            >
              <span>
                Art
              </span>
              <span>
                Institute
              </span>
              <span>
                Chicago
              </span>
            </div>
            <div
              class="title"
            >
              <h1
                class="titleText"
              >
                Exhibitions
              </h1>
              <p
                class="total"
              >
                Total:
                0
              </p>
            </div>
          </div>
        </div>
        <div
          class="nav"
        >
          <input
            alt="icon"
            src="arrowBack.svg"
            type="image"
          />
          <input
            alt="icon"
            src="arrow.svg"
            type="image"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="headerWrapper"
    >
      <div
        class="color"
      >
        <div
          class="home"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt="icon"
              class="icon"
              src="home.svg"
            />
          </a>
        </div>
        <div
          class="header"
        >
          <div
            class="logo"
          >
            <span>
              Art
            </span>
            <span>
              Institute
            </span>
            <span>
              Chicago
            </span>
          </div>
          <div
            class="title"
          >
            <h1
              class="titleText"
            >
              Exhibitions
            </h1>
            <p
              class="total"
            >
              Total:
              0
            </p>
          </div>
        </div>
      </div>
      <div
        class="nav"
      >
        <input
          alt="icon"
          src="arrowBack.svg"
          type="image"
        />
        <input
          alt="icon"
          src="arrow.svg"
          type="image"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
